### PR TITLE
feat(aapd-1195): use oauth for enter code login

### PR DESCRIPTION
**Pull request by bryn-pins**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-1195

## Description of change

Enter code to use oauth and keep tokens in session for use in api calls

## Checklist

- [x] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/appeals-service-api/src/lib/notify.js [hunk 1]
```diff
@@ -219,7 +181,7 @@ const sendSecurityCodeEmail = async (recipientEmail, code, identifier, action =
 			'Sending secure code email to appellant'
 		);
 		await NotifyBuilder.reset()
-			.setTemplateId(mapActionToTemplate(action))
+			.setTemplateId(templates.SAVE_AND_RETURN.enterCodeIntoServiceEmailToAppellant)
```
Comment (bryn-pins) [inline packages/appeals-service-api/src/lib/notify.js:184]: there is no separate lpa email so have removed this logic

File: packages/forms-web-app/src/controllers/common/enter-code.js [hunk 2]
```diff
@@ -397,7 +402,15 @@ const postEnterCodeLPA = (views) => {
 		}
 
 		// check token
-		const tokenResult = await isTokenValid(emailCode, id, user.email, req.session, user.lpaCode);
+		const enrolUsers = true;
```
Comment (beejjacobs) [inline packages/forms-web-app/src/controllers/common/enter-code.js:405]: Is this not meant to be the flag value? `enrolUsersFlag`?

File: packages/forms-web-app/src/lib/is-token-valid.js [hunk 3]
```diff
@@ -63,21 +69,68 @@ const getToken = async (token, id, emailAddress, action) => {
 
 /**
  * @param {string} token
- * @param {string} [id] - appealId or userid
+ * @param {string} [emailAddress]
+ * @param {string} [action]
+ * @returns {Promise<TokenValidResult>}
+ */
+const authToken = async (token, emailAddress, action) => {
```
Comment (beejjacobs) [inline packages/forms-web-app/src/lib/is-token-valid.js:76]: Not sure why, I find the name of this incomplete, needs a verb?

File: packages/forms-web-app/src/middleware/check-logged-in.js [hunk 4]
```diff
@@ -24,7 +24,7 @@ const checkLoggedIn = async (req, res, next) => {
 
 	const user = getAppealUserSession(req);
 
-	if (user && user?.id) {
+	if (user && user?.expiry.getTime() > Date.now()) {
```
Comment (beejjacobs) [inline packages/forms-web-app/src/middleware/check-logged-in.js:27]: Feels a much better check!

**Review by beejjacobs — COMMENTED** [review]
Code looks OK. Feels like this is important enough I should run it and check properly!

File: packages/forms-web-app/src/controllers/common/enter-code.js (hunk 2 above)
Comment (bryn-pins) [reply packages/forms-web-app/src/controllers/common/enter-code.js]: this is for lpas
the entire process relies on the sql user so this should always be true
perhaps it gets reworked if we split out appellant-dashboard and sql users into separate flags

**Review by beejjacobs — APPROVED** [review]

**Pull request merged**